From 09e8f0a32d6154a794ebbb30aef7f893d42f9883 Mon Sep 17 00:00:00 2001
From: Karthik Kambatla <kasha@cloudera.com>
Date: Wed, 24 Feb 2016 18:33:57 -0800
Subject: [PATCH 1274/2748] YARN-4729.
 SchedulerApplicationAttempt#getTotalRequiredResources
 can throw an NPE. (kasha)

(cherry picked from commit c684f2b007a4808dafbe1c1d3ce01758e281d329)
(cherry picked from commit 6a75c5af09fc1126a5406fec91da75011ca20c06)

Change-Id: I467d14f1ad95a8641c9df60e4bd40de311d5da61
---
 .../scheduler/SchedulerApplicationAttempt.java     |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/SchedulerApplicationAttempt.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/SchedulerApplicationAttempt.java
index 2b764d9..5b73cc7 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/SchedulerApplicationAttempt.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/SchedulerApplicationAttempt.java
@@ -205,7 +205,8 @@ public synchronized ResourceRequest getResourceRequest(Priority priority, String
   }
 
   public synchronized int getTotalRequiredResources(Priority priority) {
-    return getResourceRequest(priority, ResourceRequest.ANY).getNumContainers();
+    ResourceRequest request = getResourceRequest(priority, ResourceRequest.ANY);
+    return request == null ? 0 : request.getNumContainers();
   }
 
   public synchronized Resource getResource(Priority priority) {
-- 
1.7.9.5

